src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/MemoryMarshal.cs

```diff
@@ -307,7 +307,9 @@ public static bool TryGetArray<T>(ReadOnlyMemory<T> memory, out ArraySegment<T>
         {
             TManager? localManager; // Use register for null comparison rather than byref
             manager = localManager = memory.GetObjectStartLength(out _, out _) as TManager;
+#pragma warning disable 8762 // "Parameter 'manager' may not have a null value when exiting with 'true'."
```

Comment:
>#pragma warning disable 8762  [](start = 0, length = 29)

:memo: commit 11 experimented with replacing pragma with `Debug.Assert`, but that results in different IL, so I reverted.

Reply:
>but that results in different IL, so I reverted.

In release or just in debug?  Just in debug is fine :)

Reply:
Unfortunately the impact is in release.
`if (x == null) return true; else return false;` is different than `return x == null;`

[sharplab](https://sharplab.io/#v2:C4LglgNgPgAgTARgLACgYGYAE9MGFMDeqmJ2WARgPaUSYCyAFAMrABOYAdgOaYDOAlJmKkiKUuMxgAZpgZ9MAXgWYOAVwgR+wiYW06SMAOyY2qgKYBuUnvEBfG6SOYpAQwi9Le+yltA=)

It's possible that the runtime optimizes this away in assembly, but I didn't want to chance it...

Reply:
Ah, thanks, no, let's not do that :-)